utilities/backup/backup_engine_test.cc

```diff
@@ -4790,6 +4790,77 @@ TEST_F(BackupEngineTest, IOBufferSize) {
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->ClearAllCallBacks();
 }
 
+// Test stopping backup at different points in the backup lifecycle
+TEST_F(BackupEngineTest, StopBackupAtDifferentStages) {
+  const int keys_iteration = 5000;
+
+  // Test different threshold values:
+  // 0 = stop immediately (before work starts)
+  // 3, 10, 50 = stop after some work (backup may complete successfully)
+  std::vector<int> stop_thresholds = {0, 3, 10, 50};
```

Comment:
Even if this currently covers all the cases we've put in place, I don't trust that to remain true. I'd prefer the test be randomized and surface as a flaky (but true) failure than to suffer a consistent false negative (not detecting a bug). Especially since this isn't (yet) in the crash test. We should note in the PR description that adding StopBackup to the crash test is a suggested follow-up to this PR.

For randomizing the number of iterations before stopping, I suggest:
* Use something resembling a geometric distribution because smaller numbers are more likely to be interesting than specific larger numbers.
* Run several iterations per invocation of the test, ideally keeping the individual test runtime less than 3-5 seconds.